core/client/fs/src/main/java/alluxio/client/file/AlluxioFileInStream.java

```diff
@@ -448,9 +450,18 @@ boolean triggerAsyncCaching(BlockInStream stream) {
 
   private void handleRetryableException(BlockInStream stream, IOException e) {
     WorkerNetAddress workerAddress = stream.getAddress();
-    LOG.warn("Failed to read block {} of file {} from worker {}. "
-        + "This worker will be skipped for future read operations, will retry: {}.",
-        stream.getId(), mStatus.getPath(), workerAddress, e.toString());
+
+    boolean causedByClient = (e.getCause() instanceof StatusRuntimeException)
```

Comment:
Do you have a list of exception which indicates the error is from the client side not the server side?
On the other hand, what if the server would return runtime error?

Reply:
> what if the server would return runtime error?
I only encountered this one. 

> what if the server would return runtime error?
I think the server should not return a runtime error to the client, or at least it needs to convert it to an IOException.

Reply:
We need to make sure this logic would not break the previous ones, otherwise it might cause regression issue.

Reply:
Understand. Let me try to trigger one OutOfDirectMemoryError at the worker side.

Reply:
@rongrong @jja725  what's your opinion here? I remember the gRPC do some internal translation for runtime exceptions between server and client.

Reply:
I think what's implemented here is correct since the grpc stub always throw StatusRuntimeException and we wrap it into IOException. And also agree that we want to double check by triggering the same exception

Reply:
```suggestion
    boolean causedByClientOOM = (e.getCause() instanceof StatusRuntimeException)
```